app/uk/gov/nationalarchives/omega/editorial/support/DateParser.scala

```diff
@@ -44,12 +44,35 @@ import scala.util.Try
 object DateParser {
 
   private val dateTimeFormatter =
```

Comment:
Here's a suggested implementation, where we only have to parse once:

``` 
object DateParser {

  private val partDelimiter = "/"

  private val dateTimeFormatter =
    DateTimeFormatter
      .ofPattern(s"d${partDelimiter}M${partDelimiter}y")
      .withLocale(Locale.UK)
      .withResolverStyle(ResolverStyle.SMART)

  def parse(rawDateAsYearMonthDay: String): Option[LocalDate] =
    Try(LocalDate.parse(rawDateAsYearMonthDay, dateTimeFormatter)).toOption
      .filter(dateIsSameAsOriginal(_, rawDateAsYearMonthDay))

  private def dateIsSameAsOriginal(date: LocalDate, rawDateAsYearMonthDay: String): Boolean =
    dateTimeFormatter.format(date) == normaliseFormat(rawDateAsYearMonthDay)

  private def normaliseFormat(originalRawDate: String): String = {
    val partsOfOriginalRawDateWithoutLeadingZeros =
      originalRawDate
        .split(partDelimiter)
        .map(part => String.valueOf(part.toInt))
        .toSeq
    partsOfOriginalRawDateWithoutLeadingZeros.mkString(partDelimiter)
  }

}
```

Reply:
The above code snippet does't accept zero prefixed date, month and year. In the proposed implementation `LocalDate.of` would allow for this.

Reply:
Done

Reply:
Confirmed.

I know that I suggested this, but I'm now thinking that this:

```
.ofPattern("d" + partDelimiter + "M" + partDelimiter + "y")
```

or this:

```
.ofPattern(Seq("d", "M", "y").mkString(partDelimiter))
```

would be clearer than this:

```
.ofPattern(s"d${partDelimiter}M${partDelimiter}y")
```

But I'm OK with any of them, so resolving.